### ISSUE TITLE
Is it possible to register helpers at the partial level instead of global?

### ISSUE THREAD

**Issue by lowi** [wycats/handlebars.js#1334]
I want to switch from Blaze/Meteor to Handlebars/Express, and I am looking for a way to register helpers on the partial instead of the global scope.

I guess I am looking for a function Handlebars.registerPartialHelper instead of Handlebars.registerHelp.

Should I just register all my current Blaze template helpers as global helpers in Handlebars?
(I find the Blaze way very practical in terms of code isolation)

**Comment by nknapp** [comment]
No, it's not possible to do that. The helpers-object is a singleton within the Handlebars-runtime environment.
I would register all helpers as global helper. Of course, you could try to create namespaces yourself to avoid name-clashes. (by prefixing the helper-name with the partial-name)

**Comment by lowi** [comment]
I think I found a simple solution to achieve what I want by using decorators [as described here](https://www.sitepoint.com/using-inline-partials-and-decorators-with-handlebars-4-0/).

I currently have a script that generates .hbs templates from my Meteor project.
My idea is to generate a decorator helper per template from my Meteor template, and to insert an activation line in each template:

// navbar.hbs
`{{* navbarHelpers}}`

// navbarHelpers.js
```
Handlebars.registerDecorator('navbarHelpers', function(program, props, container, context) {
    container.helpers = {
        ...
    };
});

```
I will update once I have tested it.
